src/30_Advanced/Show_More_Button.html

```diff
@@ -0,0 +1,81 @@
+  <!-- We use `amp-mustache` for rendering the `amp-list` content -->
+  <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.1.js"></script>
+
+  <link rel="canonical" href="<%host%>/advanced/load_more_button/">
+  <title>Load more button</title>
+
+  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+
+</head>
+<body>
+  <!-- ## How to implement a show more button -->
+  <!-- You can implement a show more button by using `amp-list` and `amp-bind`, infact
+  `amp-list` data can be specified by using `amp-state` as a `src`. Here we are using 2 `amp-state`: `productsState`
+  represents the actual state of the list of shown products... -->
+  <amp-state id="productsState" src="/json/related_products.json">
+  </amp-state>
+
+  <!-- while the `amp-state` with id `product` is just used to implement the show-more logic where we are allowing the user to click 3 times. -->
+  <amp-state id="product">
+    <script type="application/json">
+    {
+      "moreItemsPageIndex": 0,
+      "hasMorePages": true
+    }
+    </script>
+  </amp-state>
+
+  <!-- The show more button is implemented via a form which trigger a page update by using the `submit-success` event.
+      We are then merging the form results into the items already loaded by the `amp-state`. -->
+    <form method="GET"
+        action="/json/more_related_products_page"
+        action-xhr="/json/more_related_products_page"
+        target="_top"
+        on="submit-success: AMP.setState({
+          productsState: { items: productsState.items.concat(event.response.items)},
+          product: {moreItemsPageIndex: product.moreItemsPageIndex + 1,
+                    hasMorePages: event.response.hasMorePages}
+        });">
+      <input type="hidden" name="moreItemsPageIndex" value="0" [value]="product.moreItemsPageIndex">
+      <input type="submit" value="Show more"
+        class="ampstart-btn caps m1 mb3 show"
+        [class] = "(product.hasMorePages == false ? 'hide' : 'ampstart-btn caps m1 mb3 show')">
+    </form>
+
+  <!-- We bind the `src` attribute of the `amp-list` to the id of the `amp-state` component so that the `amp-list` will use the items
```

Comment:
s/ the id of the `amp-state` component so that the `amp-list` will use the items/to the `amp-state` object containing the products.

Reply:
done